android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java

```diff
@@ -1013,6 +1031,7 @@ public void onHostPause() {
             public void onHostDestroy() {
                 dispose();
                 onDestroy();
+                reactContext.removeLifecycleEventListener(this);
```

Comment:
should this line be moved to the `dispose` method?

Reply:
Because dispose is a public method, we wouldn't want anyone calling that and then our MapView never receives lifecycle callbacks anymore... 

I chose to put it in here because this mapView instance should only ever remove lifecycle events from react whenever the host is destroyed, not whenever someone else wants to call dispose()...

Reply:
This is a good point, I agree that we should leave it out

Reply:
@nitaliano, After pondering about this further, I think that you are right. If React tells us to drop our mapView instance (```onDropViewInstance(mapView)```), and we don't remove the lifecycle listeners, we could see a nasty bug if the user background and resumes the app (hits the lifecycle callbacks).

The 'disposed' mapView instance would receive the onPause and onResume callbacks which would reenable (recreate and start) the location manager. We would not want this to happen. 

So I think we should also call ```reactContext.removeLifecycleEventListener(this);``` in the dispose method alongside calling it in ```onHostDestroy()```

Do you want to try and test this change? or should I open another PR for this?